aya/src/programs/mod.rs

```diff
@@ -805,6 +818,61 @@ impl_fd!(
     CgroupDevice,
 );
 
+/// Defines which [`Program`]s support the kernel's
+/// generic multi-prog API.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 6.6.
+pub trait MprogProgram {
+    /// Returns a borrowed reference to the file descriptor of a given
+    /// [`Program`] which has support for the kernel's generic multi-prog API.
+    fn fd(&self) -> Result<BorrowedFd<'_>, ProgramError>;
+}
+
+macro_rules! impl_mprog_fd {
+    ($($struct_name:ident),+ $(,)?) => {
+        $(
+            impl MprogProgram for $struct_name {
+                /// Returns the a borrowed reference file descriptor of this Program.
+                fn fd(&self) -> Result<BorrowedFd<'_>, ProgramError> {
+                    Ok(self.fd()?.as_fd())
+                }
+            }
+        )+
+    }
+}
+
+impl_mprog_fd!(SchedClassifier,);
+
+/// Defines which [`Link`]s support the kernel's
```

Comment:
```
Trait implemented by [Link]s which support the kernel's generic multi-prog API [link?].
```

We should at a minimum provide a link that explains what multi-prog is, or
better add some doc of our own :P

Reply:
Sounds good, I linked the commit otherwise there is docs, but I'll make some follow up issues :)